### PR TITLE
Add TypeScript typings

### LINKED ISSUE #105: Add typescript typings
The following definition works fine:

```
interface MorphDomOptions {
        getNodeKey?: (node: Node) => any,
        onBeforeNodeAdded?: (node: Node) => Node,
        onNodeAdded?: (node: Node) => Node,
        onBeforeElUpdated?: (fromEl: HTMLElement, toEl: HTMLElement) => boolean,
        onElUpdated?: (el: HTMLElement) => void,
        onBeforeNodeDiscarded?: (node : Node) => boolean,
        onNodeDiscarded?: (node : Node) => void,
        onBeforeElChildrenUpdated?: (fromEl: HTMLElement, toEl: HTMLElement) => boolean,
        childrenOnly?: boolean
}
export default function morphdom(fromNode : Node, toNode : Node | String, options?: MorphDomOptions): void;
```

It works in my own source when named "morphdom.d.ts", but I'm not sure how this could be integrated with the main morphdom source tree. I'm creating this issue so people may be able to find this.

### PR DESCRIPTION
**Pull request by aesy**
Fixes #105

### CODE REVIEW

**Comment by AutoSponge** [comment]
lgtm 👍

**Comment by ricardocasares** [comment]
+1

**Comment by JacobEvelyn** [comment]
I'd love to have this as well. Is there any reason this can't be merged?

File: typings/morphdom.d.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+	namespace morphdom {}
+
+	function morphdom(
+		fromNode: Node,
+		toNode: Node | string,
+		options?: MorphDomOptions
+	): void;
+
+	export = morphdom;
```
Comment (JacobEvelyn) [inline typings/morphdom.d.ts]: I had to change this line to `export default morphdom;` to get this to work in my project.

**Comment by benjamin-hodgson** [comment]
Any news on getting this PR merged? would be useful for us.

File: typings/morphdom.d.ts (hunk 1 above)
Comment (AutoSponge) [reply typings/morphdom.d.ts]: Does that need to be updated in this PR?

**Comment by snewcomer** [comment]
added here https://www.npmjs.com/package/@types/morphdom and #138 

published as 2.5.2!

**Pull request closed without merging**